### PR TITLE
test: add err handling in test

### PR DESCRIPTION
**Pull request by hwdef**
### Ⅰ. Describe what this PR did

add err handling in test

### Ⅱ. Does this pull request fix one issue?

NONE

### Ⅲ. Why don't you add test cases (unit test/integration test)? (你真的觉得不需要加测试吗？)

maybe need test

### CODE REVIEW

File: test/command/cmd.go [hunk 1]
```diff
@@ -150,12 +150,18 @@ func (s *Starter) Supernode(running time.Duration, args ...string) (
 	}
 
 	if err = check("localhost", environment.SupernodeListenPort, 10*time.Second); err != nil {
-		s.Kill(cmd)
+		err = s.Kill(cmd)
+		if err != nil {
```
Comment (yunfeiyanggzq) [inline test/command/cmd.go]: does this place necessary? the return is same

File: test/command/cmd.go [hunk 2]
```diff
@@ -150,12 +150,18 @@ func (s *Starter) Supernode(running time.Duration, args ...string) (
...
+			return nil, err
+		}
 		return nil, err
 	}
 	s.supernodeFileServerHome = fp.Join(dir, "repo")
 	if _, err = s.fileServer(cmd, s.supernodeFileServerHome, environment.SupernodeDownloadPort); err != nil {
-		s.Kill(cmd)
+		err = s.Kill(cmd)
```
Comment (yunfeiyanggzq) [inline test/command/cmd.go]: The same problem as the previous one

**Comment by yunfeiyanggzq** [comment]
you add some err return in function but I haven't find other functions that use this function do with the err

File: test/command/cmd.go (hunk 1 above)
Comment (hwdef) [reply test/command/cmd.go]: @yunfeiyanggzq 
I think I can rename the second error,avoid overwriting value.
what do you think?
or I delete this change,keep it the same as before.

File: test/command/cmd.go [hunk 3]
```diff
@@ -204,35 +214,53 @@ func (s *Starter) KillAll() {
...
 // kill can not be called directly, use Kill instead.
-func (s *Starter) kill(cmd *exec.Cmd) {
+func (s *Starter) kill(cmd *exec.Cmd) error {
 	if cmd == nil || cmd.Process == nil {
-		return
+		return nil
 	}
 	if cmd.ProcessState == nil {
-		cmd.Process.Signal(os.Interrupt)
+		err := cmd.Process.Signal(os.Interrupt)
```
Comment (Starnop) [inline test/command/cmd.go]:
```
if err := cmd.Process.Signal(os.Interrupt); err != nil {
	return err
}
```

Could you please use the code style as above? THX.

File: test/command/cmd.go [hunk 4]
```diff
@@ -204,35 +214,53 @@ func (s *Starter) KillAll() {
 			continue
 		}
 		if v, ok := cmd.Value.(*exec.Cmd); ok {
-			s.kill(v)
+			err := s.kill(v)
```
Comment (Starnop) [inline test/command/cmd.go]: I think we should continue even if failed to kill some cmd. And we can return the errors with a slice. WDYT?

File: test/command/cmd.go (hunk 1 above)
Comment (Starnop) [reply test/command/cmd.go]: I think it's better to update the first err. WDYT?

In addition, could you please update the scope of the err which means update the code as follows:

```
// the origin code
       if err = check("localhost", environment.SupernodeListenPort, 10*time.Second); err != nil {
		s.Kill(cmd)
		return nil, err
	}

// updated code
	if err := check("localhost", environment.SupernodeListenPort, 10*time.Second); err != nil {
		s.Kill(cmd)
		return nil, err
	}
```

**Comment by hwdef** [comment]
@Starnop 
I read all your suggestions.
thanks, and I will fix it.

File: test/command/cmd.go (hunk 3 above)
Comment (hwdef) [reply test/command/cmd.go]: @Starnop 
I don't understand this.
It is different with above.

File: test/command/cmd.go (hunk 4 above)
Comment (hwdef) [reply test/command/cmd.go]: use a slice?
It will change a lot of code.
or keep it as same as before, do not handle this error.WDYT

Comment (Starnop) [reply test/command/cmd.go]: IMOP, that's OK. 😄

File: test/command/cmd.go (hunk 3 above)
Comment (Starnop) [reply test/command/cmd.go]: I mean you can update your code style

from
```
err := cmd.Process.Signal(os.Interrupt)
if err != nil {
    return err
}
```

to
```
if err := cmd.Process.Signal(os.Interrupt); err != nil {
	return err
}
```

😄

**Comment by hwdef** [comment]
@Starnop 
I deleted all controversial changes.

File: test/command/cmd.go [hunk 5]
```diff
@@ -210,11 +210,12 @@ func (s *Starter) KillAll() {
 }
 
 // Clean cleans all temporary directories and processes.
-func (s *Starter) Clean() {
+func (s *Starter) Clean() error {
```
Comment (yeya24) [inline test/command/cmd.go]: Do you still need the return value `error`? I see here only return nil

Comment (hwdef) [reply test/command/cmd.go]: @yeya24 
You are right,I changed it.

**Comment by Starnop** [comment]
LGTM.

**Pull request merged**